apps/roam/src/utils/renderNodeTagPopup.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState, useRef, useEffect } from "react";
+import ReactDOM from "react-dom";
+import { Button, Popover, Position } from "@blueprintjs/core";
+
+let currentPopup: HTMLDivElement | null = null;
+
+export const removeNodeTagPopup = () => {
+  if (currentPopup) {
+    ReactDOM.unmountComponentAtNode(currentPopup);
+    currentPopup.remove();
+    currentPopup = null;
+  }
+};
+
+const HoverControlledPopover = ({
+  onClick,
+  label,
+  onClose,
+  tagElement,
+}: {
+  onClick: () => void;
+  label: string;
+  onClose: () => void;
+  tagElement: HTMLElement;
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const timeoutRef = useRef<number>();
+
+  useEffect(() => {
```

Comment:
None of these listeners should be required.  If you rewatch our last meeting you'll see that just the initial observer code + popover will handle this.